clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -463,34 +463,49 @@ public ReadyCheckResult ready(Cluster cluster, long nowMs) {
         boolean exhausted = this.free.queued() > 0;
         for (Map.Entry<TopicPartition, Deque<ProducerBatch>> entry : this.batches.entrySet()) {
             Deque<ProducerBatch> deque = entry.getValue();
+
```

Comment:
If I read the change correctly, the remaining logic of processing the batch is unchanged, we only refactor the processing part out of the synchronized session?

Basically I'm wondering if we should make this default behavior, or add a config to make it opt-in for users who want such improvement.

Reply:
yes, that's how I understand it.

The reason for making it opt-in is to minimize blast radius in case this change is breaking in some way?